### ISSUE TITLE
Pathwise sampling

### ISSUE THREAD

**Issue by rossviljoen** [rossviljoen/SparseGPs.jl#7]
It would be nice to eventually add pathwise posterior sampling from [1].

Essentially, reimplement https://github.com/aterenin/SparseGaussianProcesses.jl but using AbstractGPs etc.

[1] Wilson, James, et al. "Efficiently sampling functions from Gaussian process posteriors." International Conference on Machine Learning. PMLR, 2020. https://arxiv.org/abs/2002.09309

**Comment by willtebbutt** [comment]
Yeah. I spoke with Alex the other day. We could rope him in to this at some point.

**Comment by theogf** [comment]
I already implemented it in another project, I could also help

**Comment by rossviljoen** [comment]
I've implemented a basic version of this [here](https://github.com/rossviljoen/PathwiseSampling.jl)

**Comment by theogf** [comment]
This sounds great, for the RFF, I would not like to clutter KernelFunctions even more. Maybe we could have an additional package which deals strictly with the whole Fourier representation of kernels.
